src/Microsoft.AspNet.Mvc.Razor/TagHelpers/UrlResolutionTagHelper.cs

```diff
@@ -0,0 +1,212 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Microsoft.Framework.WebEncoders;
+
+namespace Microsoft.AspNet.Mvc.Razor.TagHelpers
+{
+    /// <summary>
+    /// <see cref="ITagHelper"/> implementation targeting elements containing attributes with URL expected values.
+    /// </summary>
+    /// <remarks>Resolves application relative URLs that are not targeted by other <see cref="ITagHelper"/>s. Runs
+    /// prior to other <see cref="ITagHelper"/>s to ensure application-relative URLs are resolved.</remarks>
```

Comment:
As discussed elsewhere, "application relative" is not consistent with how that term is used elsewhere. But also have no existing doc comments hyphenating "application-relative".